Contacts CRUD

nop33 opened the pull request:
Sorry for the long PR, the feature of contacts is a big one. I will work on the remaining sub-features (like ~contact sharing,~ selecting contact in send screens #160, etc) in a follow-up PR.

mvaivre commented on src/persistent-storage/wallets.ts:
```diff
@@ -260,6 +262,70 @@ export const persistAddressesMetadata = async (walletId: string, addressesMetada
   await persistWalletsMetadata(walletsMetadata)
 }
 
+export const persistContact = async (contactData: ContactFormData) => {
```
Nit: why not adding those function to a dedicated `contacts.ts` file?

nop33 replied:
Good question. I chose to leave them in this file because I made `contacts` be part of the wallet metadata object. Since all wallet metadata functions are in this file, I decided to keep these here as well. Not a strong opinion on this.

mvaivre commented:
Functional testing: I get an error when trying to create a contact.

nop33 commented:
> Functional testing: I get an error when trying to create a contact.

I fixed this, it was probably cause the wallet you used had old metadata stored. I updated the code to take this into consideration.

mvaivre commented on src/screens/Addresses/Contact/EditContactScreen.tsx:
```diff
@@ -0,0 +1,110 @@
+                  setIsDeleting(true)
+
+                  try {
+                    await deleteContact(params.contactId)
```
This function call should always be followed by the dispatch of `contactDeletedFromPersistentStorage` right? In this case, why not dispatch directly from within the `deleteContact` function?

mvaivre commented on src/screens/Addresses/Contact/EditContactScreen.tsx:
```diff
@@ -0,0 +1,110 @@
+    setLoading(true)
+
+    try {
+      await persistContact(formData)
```
Same remark as above.

mvaivre commented on src/screens/Addresses/Contact/NewContactScreen.tsx:
```diff
@@ -0,0 +1,69 @@
+    try {
+      const id = await persistContact(formData)
+
+      dispatch(contactStoredInPersistentStorage({ ...formData, id }))
```
Here too (I won't signal the others if any)

nop33 commented on src/components/ListItem.tsx:
```diff
@@ -0,0 +1,121 @@
+  innerStyle?: StyleProp<ViewStyle>
+  hideSeparator?: boolean
+  rightSideContent?: ReactNode
+  children?: ReactNode
```
It does, but if I remove it I get this error:

```
    Type 'ReactNode | ((state: PressableStateCallbackType) => ReactNode)' is not assignable to type 'ReactNode'.ts(2769)
```

nop33 commented:
I've addressed all your comments in a follow-up PR, is that a problem for you?https://github.com/alephium/mobile-wallet/pull/167/commits

mvaivre reviewed (approved):
🏄‍♂️ 👏
